### PR TITLE
Beeps

### PR DESCRIPTION
**Pull request by Sv3n**

### CODE REVIEW

File: pypopquiz/backends/moviepy.py [hunk 1]
```diff
@@ -16,6 +21,46 @@ def silence(_) -> float:
...
+            selector1 = interval[1] > t
+            selector = np.logical_and(selector0, selector1)
+
+            t_offset = t - interval[0]
+            tone = np.array(np.sin(freq_hz * np.pi * t_offset))
+
+            channels_out = []
+            channels = original.T
+            for channel in channels:
+                channels_out.append(np.where(selector, tone, channel))
```
Comment (CNugteren) [inline pypopquiz/backends/moviepy.py]: A bit more concise perhaps:
```
channels_out = [np.where(selector, tone, channel) for channel in original.T]
```

File: pypopquiz/backends/moviepy.py [hunk 2]
```diff
@@ -16,6 +21,46 @@ def silence(_) -> float:
...
+    def beeped(get_frame: Callable, time_seconds: Union[float, np.array]):
+        """Fx callback that replaces the original sounds with a tone in an interval."""
+        t = time_seconds  # pylint: disable=invalid-name
+        original = get_frame(t)
+
+        if np.isscalar(t):
+            t = typing.cast(float, t)  # pylint: disable=invalid-name
+            if interval[0] < t < interval[1]:
+                t_offset = t - interval[0]
+                out = [np.sin(freq_hz * np.pi * t_offset)]
```
Comment (CNugteren) [inline pypopquiz/backends/moviepy.py]: You can already return here and in line 45 and 59 as well, then you don't need all the else statements, right? Makes the logic a little bit easier to follow perhaps.

File: pypopquiz/io.py [hunk 3]
```diff
@@ -31,6 +44,54 @@ def total_duration(clips: List[Dict]) -> int:
     return sum((get_interval_duration(item["interval"]) for item in clips))
 
 
+def substitute_variables(data: Dict) -> None:
```
Comment (CNugteren) [inline pypopquiz/io.py]: Maybe we can put all the variables related functions in a separate file to make the I/O file a bit less crowded?

File: pypopquiz/backends/moviepy.py [hunk 4]
```diff
@@ -162,13 +232,27 @@ def add_audio(self, other: 'Moviepy') -> None:  # type: ignore
...
 
-        if isinstance(other.clip, moviepy.editor.AudioClip):
+        if isinstance(other.clip, med.AudioClip):
             audio = other.clip
         else:
             audio = other.clip.audio
 
         self.clip = self.clip.set_audio(audio)
 
+    def audio_normalize(self) -> None:
```
Comment (CNugteren) [inline pypopquiz/backends/moviepy.py:237]: Do you use this somewhere or is it just convenient to have for later?

File: samples/beep_test.json [hunk 5]
```diff
@@ -0,0 +1,30 @@
+{
```
Comment (CNugteren) [inline samples/beep_test.json]: Was this file supposed to go in? If so, should we rename it in the same pattern as the other JSONs? Or perhaps we should actually also rename the other JSON sampes to give more descriptive names.

File: samples/beep_test.json [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+  "round": 7,
+  "theme": "Piepjes",
```
Comment (CNugteren) [inline samples/beep_test.json]: Should we make it English? :-)

File: pypopquiz/io.py [hunk 7]
```diff
@@ -156,6 +217,7 @@ def verify_input(input_data: Dict) -> None:
         }
     }
     jsonschema.validate(input_data, schema)
```
Comment (CNugteren) [inline pypopquiz/io.py:179]: You didn't change anything in the JSON schema? New things are now possible, right? Although optional, you can still describe their expected types and sub-types.

File: pypopquiz/backends/moviepy.py (hunk 4 above)
Comment (Sv3n) [reply pypopquiz/backends/moviepy.py:237]: I think it will come in useful later. I added it because I didn't want to forget about it.

**Review by CNugteren — APPROVED** [review]

**Pull request merged**